build-lib/util/recursive-write-file.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+const fs = require('fs');
+const path = require('path');
+
+function ensureDirExists (fullPath) {
+
+  if (!fs.existsSync(fullPath)) {
+    if (ensureDirExists(path.dirname(fullPath))) {
+      fs.mkdirSync(fullPath);
+    } else {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+module.exports = function (fullPath, content) {
+  if(ensureDirExists(path.dirname(fullPath))) {
+    fs.writeFileSync(fullPath, content);
+    return true;
+  }
+
+  return false;
+};
```

Comment:
What is recursive about this? I feel like you should just have a `createDirectoryIfNoneExists` function that takes the path, then write files as normal.